dataflow/src/domain/mod.rs

```diff
@@ -1566,6 +1574,80 @@ impl Domain {
         }
     }
 
+    fn handle_recovery(&mut self) {
+        let node_info: Vec<_> = self.nodes
+            .iter()
+            .map(|(index, node)| {
+                let n = node.borrow();
+                (index.clone(), n.global_addr(), n.is_transactional())
+            })
+            .collect();
+
+        for (local_addr, global_addr, is_transactional) in node_info {
+            let checktable = self.transaction_state.get_checktable();
+            let path = self.persistence_parameters.log_path(
+                &local_addr,
+                self.index,
+                self.shard.unwrap_or(0),
+            );
+
+            let file = match File::open(&path) {
+                Ok(f) => f,
+                Err(ref e) if e.kind() == ErrorKind::NotFound => {
+                    warn!(
+                        self.log,
+                        "No log file found for node {}, starting out empty",
+                        local_addr
+                    );
+
+                    continue;
+                }
+                Err(e) => panic!("Could not open log file {:?}: {}", path, e),
+            };
+
+            BufReader::new(file)
+                .lines()
+                .filter_map(|line| {
+                    let line = line.unwrap();
```

Comment:
Can you make this an `expect` instead, and then state why we're expecting it to not fail?

Reply:
Alternatively, given that you're already using `filter_map`, just return `None` if `line.is_err()`. Something like:
```rust
line.ok().and_then(|line| -> Option<Vec<Records>> { serde_json::from_str(&line).ok() })
```

Reply:
I changed it to an `.expect` as I thought it might be good to blow up in the case that this ever happens (IO-errors, or invalid UTF-8 in the file - if I understand it correctly).
Let me know if you disagree though.

https://github.com/mit-pdos/distributary/pull/37/files#diff-f3207323d6f34ad3c124fbcb4cb87a40R1612